docs/concepts/workloads/controllers/deployment.md

```diff
@@ -658,18 +719,29 @@ $ echo $?
 All actions that apply to a complete Deployment also apply to a failed Deployment. You can scale it up/down, roll back
 to a previous revision, or even pause it if you need to apply multiple tweaks in the Deployment pod template.
 
-## Use Cases
+## Clean up Policy
+
+You can set `.spec.revisionHistoryLimit` field in a Deployment to specify how many old ReplicaSets for
+this Deployment you want to retain. The rest will be garbage-collected in the background. By default,
+all revision history will be kept. In a future version, it will default to switch to 2. 
+
+**Note:** Explicitly setting this field to 0, will result in cleaning up all the history of your Deployment
+thus that Deployment will not be able to roll back.
+
+
+## Use Cases 
 
 ### Canary Deployment
 
-If you want to roll out releases to a subset of users or servers using the Deployment, you can create multiple Deployments, one for each release,
-following the canary pattern described in [managing resources](/docs/concepts/cluster-administration/manage-deployment/#canary-deployments).
+If you want to roll out releases to a subset of users or servers using the Deployment, you
+can create multiple Deployments, one for each release, following the canary pattern described in
+[managing resources](/docs/concepts/cluster-administration/manage-deployment/#canary-deployments).
 
 ## Writing a Deployment Spec
 
-As with all other Kubernetes configs, a Deployment needs `apiVersion`, `kind`, and
-`metadata` fields.  For general information about working with config files,
-see [deploying applications](/docs/user-guide/deploying-applications), [configuring containers](/docs/user-guide/configuring-containers), and [using kubectl to manage resources](/docs/user-guide/working-with-resources) documents.
+As with all other Kubernetes configs, a Deployment needs `apiVersion`, `kind`, and `metadata` fields.
+For general information about working with config files, see [deploying applications](/docs/user-guide/deploying-applications),
+[configuring containers](/docs/user-guide/configuring-containers), and [using kubectl to manage resources](/docs/user-guide/working-with-resources) documents.
```

Comment:
* Find a replacement link for configuring containers under [Tasks](https://kubernetes.io/docs/tasks/) or remove it altogether -- I don't currently see an appropriate task to link to.
* Update using kubectl... link to `/docs/tutorials/object-management-kubectl/object-management/`